connector/src/main/scala/com/microsoft/kusto/spark/datasource/KustoRelation.scala

```diff
@@ -38,79 +38,77 @@ private[kusto] case class KustoRelation(kustoCoordinates: KustoCoordinates,
         val schema = StructType.fromDDL(customSchema.get)
         KustoSchema(schema, Set())
       }
-      else getSchema
+      else {
+        getSchema
+      }
     }
     cachedSchema.sparkSchema
   }
 
   override def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row] = {
-    val kustoClient = KustoClientCache.getClient(kustoCoordinates.clusterUrl, authentication, kustoCoordinates.ingestionUrl, kustoCoordinates.clusterAlias)
-    var timedOutCounting = false
-    val forceSingleMode = readOptions.readMode.isDefined && readOptions.readMode.get == ReadMode.ForceSingleMode
-    var useSingleMode = forceSingleMode
-    var res: Option[RDD[Row]] = None
-    if (readOptions.readMode.isEmpty){
-      var count = 0
-      try {
-        count = KDSU.
-          estimateRowsCount(kustoClient.engineClient, query, kustoCoordinates.database, clientRequestProperties.orNull)
-      } catch {
-        // Assume count is high if estimation got timed out
-        case e: Exception =>
-          if (forceSingleMode) {
-            // Throw in case user forced LeanMode
-            throw e
-          }
-          // By default we fallback to distributed mode
-          timedOutCounting = true
-      }
-      if (count == 0 && !timedOutCounting) {
-        res = Some(sparkSession.emptyDataFrame.rdd)
-      } else {
-        // Use distributed mode if count is high or in case of a time out
-        useSingleMode =  !(timedOutCounting || count > KustoConstants.DirectQueryUpperBoundRows)
-      }
+    val kustoClient = KustoClientCache.getClient(kustoCoordinates.clusterUrl, authentication,
+      kustoCoordinates.ingestionUrl, kustoCoordinates.clusterAlias)
+    val isUserOptionForceSingleMode = readOptions.readMode.contains(ReadMode.ForceSingleMode)
+    val storageParameters = maybeStorageParameters.getOrElse(kustoClient.getTempBlobsForExport)
```

Comment:
please dont call kustoClient.getTempBlobsForExport if SingleMode is chosen (by user or us)